tests/shutdown/shutdown.pm

```diff
@@ -29,4 +29,18 @@ sub test_flags {
     return {fatal => 1};
 }
 
+sub post_fail_hook {
+    my ($self) = shift;
+    $self->SUPER::post_fail_hook;
+
+    if (get_var('DEBUG_SHUTDOWN')) {
+        # Export logs after failure
+        select_console('log-console');
+        assert_script_run("journalctl --no-pager -b 0 > /tmp/full_journal.log");
+        upload_logs "/tmp/full_journal.log";
+        assert_script_run("dmesg > /tmp/shutdown.log");
```

Comment:
When `DEBUG_SHUTDOWN` is set we will already put dmesg output on the serial, by putting script to `/usr/lib/systemd/system-shutdown/`. Also, I doubt that any of these will work during the shutdown. Please, ask @OleksandrOrlov, as he has done everything what is possible there I guess...

Reply:
Yes, in case if one need to see the shutdown logs, just add DEBUG_SHUTDOWN=1 to the Test Suite and run the job. The logs will be stored in serial0.txt.

Reply:
good to know that dmesg is saved in serial0.txt, but how about journal log? 
I want to introduce this step extra because we can add other logs later which might be helpful.

bootloader_setup.pm:

if (get_var('DEBUG_SHUTDOWN')) {
        $args .= " systemd.log_level=debug systemd.log_target=kmsg log_buf_len=1M printk.devkmsg=on enforcing=0 plymouth.enable=0";
    }

Thanks to @rwx788 and @OleksandrOrlov for your time and comments!